provider/unique_ptr.hpp

```diff
@@ -180,6 +179,65 @@ unique_ptr_move<T, D>::unique_ptr_move (
 }
 
 
+// unique_ptr_move (array specialization)
+//------------------------------------------------------------------------------
+template<typename T, typename D>
+struct unique_ptr_move<T[], D>
```

Comment:
There shouldn't have to be a partial specialization for arrays for this class.  I know that there is a way to do this without one.  There should be a way for unique_ptr_move to typename the correct unique_ptr to avoid this code duplication.
Writing a specialization for this violates the idea of generic programming.